### DISCUSSION TITLE
[Spec] Introduce Actions and remove Events and Commands properties from controls

### DISCUSSION THREAD

**Discussion by vhugogarcia** [dotnet/maui#22923] [Ideas]
# Introducing Actions for .NET MAUI controls

My main motivation is to reduce the confusion and merge the user actions on an element within a screen. Currently, as an example on Xamarin.Forms a button have properties for clicked, and additionally it has command and command parameter and then we have tap gestures, so, 3 times the same actions. It will be ideal to combine all those properties in something better that works not only for events but for commands in a more granular and scalable way. 

This is just my idea! This is the first time I write a [Spec], I'm sorry if I'm missing something. And please feel free to share feedback.

# Scenarios

## C# Example
```csharp
var thing = new Button();
thing.Actions = new({
    Tap = new TapGesture(Command, CustomCommandParameter, 2), // Command, command parameter, number of taps
    TouchAndHold = new TouchAndHoldGesture(EventName, CustomEventParameters), // A custom event name, event parameters
});

```

## XAML Example
```xaml
<Button x:Name="thing">
    <Button.Actions>
           <TapGesture ActionName="{Binding SignIn}" ActionParameter="{.}" />
           <TouchAndHoldGesture ActionName="Share_click_event" ActionParemeter="hello world!" />
    </Button.Actions>
</Button>
```

# Backward Compatibility

The legacy Command, CommandParameter and Events of controls will remain, but can me marked as [deprecated]. Also the <TapGestures> can be deprecated and introduce a new one merged and completed.

# Difficulty : maybe high?

**Comment by ghost** [comment]
We've moved this issue to the Future milestone. This means that it is not going to be worked on for the coming release. We will reassess the issue following the current release and consider this item at that time.
